qiita_pet/support_files/doc/source/dev/plugins.rst

```diff
@@ -0,0 +1,81 @@
+.. _plugins:
+
+.. index :: plugins
+
+Plugins
+=======
+
+Qiita plugins allow to extend the functionality of Qiita by adding new data
+formats and/or new processing pipelines. The list of officially supported
+plugins can be found in the `Qiita Spots <https://github.com/qiita-spots>`__
+GitHub organization.
+
+Overview of Qiita's plugin system
+---------------------------------
+
+Qiita's plugin system has been designed to increase Qiita's flexibility to
+process -omics datasets, while allowing plugin developers to create plugins
+using the programming language that they are more comfortable with.
+
+In order to achieve this flexibility, Qiita represents the different -omics
+datasets using ``Artifacts``. An ``Artifact`` is a file or set of files that
+conceptually represents a dataset. However, Qiita does not understand the
+contents of the artifact, so it doesn't know to show information about the
+artifact or which pipelines/methods can be applied to it. It is up to the Qiita
+plugins to define what an artifact is and which methods can be applied to it.
+With the goal of reducing community development effort, there are two different
+types of plugins: ``Qiita Type Plugins`` and ``Qiita Plugins``.
+
+The ``Qiita Type Plugins`` define new artifact types. These plugins must define
+two operations: ``Validate`` and ``Generate HTML summary``.
+
+Given the artifact type, the prep information and the user-uploaded files, the
+``Validate`` operation decides if the user-uploaded files creates a valid
+artifact of the given type. If the files do not create a valid artifact, the
+plugin should try to fix the files using the given information, but only if the
+files can be deterministically fixed and not generate ambiguous results.
+
+The ``Generate HTML summary`` generates a single HTML file with an overview
+of the contents of the artifact. The only limitation on this HTML is that all
```

Comment:
I don't really see it as a limitation, what about?
Note that the summary HMTL is a single file and should contain all information about all the files attached to that Artifact.